x/ibc/mock/bank/internal/keeper/keeper.go

```diff
@@ -0,0 +1,134 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc"
+	chantypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	ics23 "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+	"github.com/cosmos/cosmos-sdk/x/ibc/mock/bank/internal/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+type Keeper struct {
+	cdc  *codec.Codec
+	key  sdk.StoreKey
+	ibck ibc.Keeper
+	bk   types.BankKeeper
+}
+
+func NewKeeper(cdc *codec.Codec, key sdk.StoreKey, ibck ibc.Keeper, bk types.BankKeeper) Keeper {
+	return Keeper{
+		cdc:  cdc,
+		key:  key,
+		ibck: ibck,
+		bk:   bk,
+	}
+}
+
+// SendTransfer handles transfer sending logic
+func (k Keeper) SendTransfer(ctx sdk.Context, srcPort, srcChan string, amount sdk.Coin, sender sdk.AccAddress, receiver string, source bool, timeout uint64) sdk.Error {
+	// get the port and channel of the counterparty
+	channel, ok := k.ibck.ChannelKeeper.GetChannel(ctx, srcPort, srcChan)
+	if !ok {
+		return sdk.NewError(sdk.CodespaceType(types.DefaultCodespace), chantypes.CodeChannelNotFound, "failed to get channel")
+	}
+
+	dstPort := channel.Counterparty.PortID
+	dstChan := channel.Counterparty.ChannelID
+
+	// get the next sequence
+	sequence, ok := k.ibck.ChannelKeeper.GetNextSequenceSend(ctx, srcPort, srcChan)
```

Comment:
This should be able to use an unordered channel, so this should be unnecessary

Reply:
It seems that the same logic is applied to sendPacket in ICS4，regardless of whether the channel is ordered or unordered.

Reply:
> This should be able to use an unordered channel, so this should be unnecessary

It seems that the same logic is applied to sendPacket in ICS4，regardless of whether the channel is ordered or unordered.

Reply:
Ah yes - a unique sequence is required, it doesn't need to be sequential, but this is also fine.